src/mon/OSDMonitor.cc

```diff
@@ -9540,6 +9540,11 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       err = -ENOENT;
       goto reply;
     }
+    if (pending_inc.old_pools.count(pgid.pool())) {
+      ss << "pool of " << pgid << " is pending removal";
+      err = -ENOENT;
+      goto reply;
```

Comment:
instead of goto reply this should wait for the commit and then return the error (the pool deletion hasn't committed yet)

Reply:
fixed